src/lib/storage/zero_suppression/simd_bp128/simd_bp128_decoder.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <array>
+#include <memory>
+#include <numeric>
+#include <utility>
+
+#include "storage/zero_suppression/base_zero_suppression_decoder.hpp"
+
+#include "oversized_types.hpp"
+#include "simd_bp128_packing.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+class SimdBp128Vector;
+
+/**
+ * @brief Implements point-access into a SIMD-BP128 compressed vector
+ *
+ * The decoder caches the last decoded block and meta-block.
+ * Performs best if the values are accessed in a cache-coherent manner, i.e.,
+ * sequentially without too many jumps backwards.
+ */
+class SimdBp128Decoder : public BaseZeroSuppressionDecoder {
+ public:
+  using Packing = SimdBp128Packing;
+
+ public:
+  explicit SimdBp128Decoder(const SimdBp128Vector& vector);
+  SimdBp128Decoder(const SimdBp128Decoder& other);
+
+  SimdBp128Decoder(SimdBp128Decoder&& other) = default;
+  ~SimdBp128Decoder() = default;
+
+  uint32_t get(size_t i) final {
+    if (_is_index_within_cached_block(i)) {
+      return _get_within_cached_block(i);
+    }
+
+    if (_is_index_within_cached_meta_block(i)) {
+      return _get_within_cached_meta_block(i);
+    }
+
+    if (_is_index_after_cached_meta_block(i)) {
+      const auto relative_index = _index_within_cached_meta_block(i);
+      const auto relative_meta_block_index = relative_index / Packing::meta_block_size;
+
+      _read_meta_info_from_offset(relative_meta_block_index);
+      return _get_within_cached_meta_block(i);
+    }
+
+    _reset_cached_meta_block();
```

Comment:
You don't reset the block, you reset the cache. How about `_clear_meta_block_cache`?

Reply:
Good point.